src/StorageFlattening.cpp

```diff
@@ -256,12 +260,16 @@ class FlattenDimensions : public IRMutator {
         vector<ProvideValue> values;
         flatten_provide_values(values, provide);
 
+        bool is_output = false;
+        for (Function f : outputs) {
+            is_output |= f.name() == provide->name;
```

Comment:
Do  you need to iterate through all outputs or can you break after finding the match?